pkg/controllers/managementapi/usercontrollers/usercontroller_test.go

```diff
@@ -0,0 +1,139 @@
+package usercontrollers
```

Comment:
Overall test nit:

You have several different function for the different test cases. You could probably make these into one function and just run through the different cases. Up to you though.

Reply:
I would also recommend adding some error cases (particularly for cases like I can't update the annotation or I can't stop the cluster).